src/AutoMapper/QueryableExtensions/ExpressionBuilder.cs

```diff
@@ -268,45 +268,12 @@ private ExpressionResolutionResult ResolveExpression(IMemberMap propertyMap, Typ
                 return null;
             }
             result = matchingExpressionConverter.GetExpressionResolutionResult(result, propertyMap, letPropertyMaps);
-            if(propertyMap.NullSubstitute != null && result.Type.IsNullableType())
+            if(propertyMap.NullSubstitute != null && result.ResolutionExpression is MemberExpression && (result.Type.IsNullableType() || result.Type == typeof(string)))
```

Comment:
Perhaps remove the type check altogether? And simply not use NullSubstitute when it doesn't make sense.